### PR TITLE
Move all utility packages from js-threads

### LINKED ISSUE #817: Mirgate: Move @textile/grpc-transport over here

### PR DESCRIPTION
**Pull request by carsonfarmer**
## Description

This is the precursor to moving everything over from js-threads into this mono-repo. This includes all utility libraries. It does not include threads-client (see #757), which currently requires its own testing env that doesn't quite match what we have here.

Note that this PR includes a ton of commits, to preserve (well, mostly) the commit history of all packages/updates in js-threads.

Fixes #817 

## Type of change

Administrative

## How Has This Been Tested?

Should not affect tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

> This PR template comes from https://github.com/embeddedartistry/templates

### CODE REVIEW

File: package.json
Comment (carsonfarmer) [inline package.json:7]: faster cleanup and remove this "extra" dist at the top level that is created by the esm build step

File: packages/admin/tsconfig.json
Comment (carsonfarmer) [inline packages/admin/tsconfig.json:8]: all sub-packages have now been updated with their references, so all local references should be up to date

File: packages/context/README.md
Comment (carsonfarmer) [inline packages/context/README.md:1]: I've switching all incoming readmes to use the "pattern". but note, we should decide if @textile/hub should still be the main entry point for everything, or if it should be something like @textile/textile?

Comment (carsonfarmer) [reply packages/context/README.md:1]: Good discussion point @andrewxhill

File: packages/context/package.json
Comment (carsonfarmer) [inline packages/context/package.json:21]: these will have to be updated

**Review by carsonfarmer — COMMENTED** [review]
A lot of administrative changes here. You can probably ignore most. See my comments where there is some room for discussion. For instance, I haven't moved for @textile/threads or @textile/threads-client yet...

File: packages/context/README.md
Comment (andrewxhill) [reply packages/context/README.md:1]: what would textile/textile be?

Comment (andrewxhill) [reply packages/context/README.md:1]: {textile}/hub <- textile already is in the namespace. i think all packages should be designed to work with the hub. you can use them in a mode that wont touch the hub, but that's an option the developer chooses. they are still hub packages none-the-less.

**Review by andrewxhill — APPROVED** [review]
this is great. i think this might make our lives a bit easier every week!

File: packages/context/README.md
Comment (carsonfarmer) [reply packages/context/README.md:1]: yup, makes sense. @textile/hub it is!

**Comment by carsonfarmer** [comment]
Note that I am _not_ going to squash this before merging in order to preserve the long history from js-threads.

**Pull request merged**
